src/pages/Test-Rig-Two/Story/hangman-game/HangmanGame.jsx

```diff
@@ -0,0 +1,378 @@
+
+  const [showSplashScreen, setShowSplashScreen] = useState(
+    localStorage.getItem('hasShownSplashScreen') === 'true' ? true : false
+  );
+
+  useEffect(() => {
+    targetWords.map((word, index) => {
+      return (
+        <React.Fragment key={index}>
+          {words.push(word?.contentSourceData[0]?.text)}
+        </React.Fragment>
+      );
+    });
+  }, [words]);
+
+  useEffect(() => {
+    resetGame();
+  }, []);
+
+  const resetGame = () => {
+    const randomWord = words[currentWordIndex];
+    setWord(randomWord?.toLowerCase());
+    // const guessWordSplit = randomWord?.split('');
+    setGuessedWord(
+      Array(
+        splitGraphemes(randomWord)?.filter(
+          item =>
+            item !== '‌' &&
+            item !== '' &&
+            item !== ' ' &&
+            item !== '-' &&
+            item !== '.'
+        )?.length
+      ).fill('_')
+    );
+
+    setHint(randomWord?.length - 1);
+    setIncorrectGuesses(0);
+    setGameWon(false);
+    setGameLost(false);
+    setCurrentLine(0);
+    setHintText('');
+    setHintCharArray('');
+  };
+
+  const handleGuess = letter => {
+    if (!gameWon && !gameLost) {
+      if (word?.includes(letter)) {
+        setGuessedWord(prevGuessedWord =>
+          splitGraphemes(word)?.map((char, index) =>
+            char === letter ? letter : prevGuessedWord[index]
+          )
+        );
+      } else {
+        setIncorrectGuesses(prevIncorrectGuesses => prevIncorrectGuesses + 1);
+      }
+    }
+  };
+
+  const handleNextWord = () => {
+    setCurrentWordIndex(prevIndex => prevIndex + 1);
+    // setCurrentLine(prevIndex => prevIndex + 1);
+  };
+
+  const provideHint = () => {
+    const hints = splitGraphemes(word).filter(
+      item => item !== '‌' && item !== '' && item !== ' ' && item !== '-'
+    );
+
+    if (currentLine < hints.length - 1) {
+      setCurrentLine(prevVal => prevVal + 1);
+      setHint(prevValue => prevValue - 1);
+    } else {
+      toast({
+        position: 'top',
+        title: `Oops! You've used all available hints`,
+        duration: 2000,
+        status: 'warning',
+      });
+      return;
+    }
+    setHintText(word);
+    setHintCharArray(hints);
+  };
+
+  // Function to merge two arrays and shuffle the result
+
+  useEffect(() => {
+    if (guessedWord.join('') !== '' && guessedWord.join('') === word) {
+      setGameWon(true);
+    } else if (incorrectGuesses + 1 >= maxIncorrectGuesses) {
+      setGameLost(true);
+    }
+  }, [guessedWord, word, incorrectGuesses, maxIncorrectGuesses]);
+
+  const renderWord = () => (
+    <HStack spacing="5">
+      {guessedWord.map((letter, index) => (
+        <Text
+          key={index}
+          fontSize={40}
+          bg={gameWon && letter !== '_' ? 'green.100' : 'transparent'}
+        >
+          {letter}
+        </Text>
+      ))}
+    </HStack>
+  );
+
+  const renderAlphabet = () => {
+    const alphabetRowsEnglish = ['qweiop'];
```

Comment:
why this is hardcoded

Reply:
random char should show here for keyboard

Reply:
This is also Resolved